src/lib/dialog/dialog-ref.ts

```diff
@@ -35,4 +36,29 @@ export class MdDialogRef<T> {
   afterClosed(): Observable<any> {
     return this._afterClosed.asObservable();
   }
+
+  /**
+   * Updates the dialog's dimensions.
+   * @param width New width of the dialog.
+   * @param height New height of the dialog.
+   * @param position New position of the dialog.
+   */
+  updateDimensions(width?: string, height?: string, position?: DialogPosition): void {
```

Comment:
Hm it feels to me like position should be in a separate method, like `updatePosition()`. If you just want to update the position, having to write `null, null` as the first two args feels a bit awkward.  If we separate, we could make it more fluent, too:

```ts
dialog.updateDimensions('100px', '100px')
      .updatePosition({top: '0', left: '0'}) 
```